### DISCUSSION TITLE
Error validating two equal values

### DISCUSSION THREAD

**Discussion by leandrodavimg** [adonisjs/core#2120] [General]
I am doing a validation of the ID passed in the route with the user ID logged in.

validation is always returning true even though it is false

```js

      const { id }: UserProps = params
      console.log(`ID parameter: ${id}`)
      console.log(`ID AUTH: ${auth.user?.id}`)
      console.log(id !== auth.user?.id)

      if (id !== auth.user?.id) {
        return response.status(403).send({
          message: 'not authorized',
        })
      }
```
in my route: /user/edit/1
in my user logged: {"id": 1, ...}

My return in terminal
```bash
ID parameter: 1
ID AUTH: 1
true
```

I created the following type for my parameter
```js
interface UserProps {
  id: number
}
```

maybe it's simple to solve, but my head is crazy about it

**Comment by steschwa** [comment]
Not quite sure what the problem is.

You check for inequality `if (id !== auth.user?.id)`. So your `if` block is never executed.
Therefore if you're logged in as User:1 and request `/user/edit/1` the response should be an empty **200 OK**.

**Reply by leandrodavimg** [reply]
The return on my api is true.
```json
{
  "message": "not authorized"
}
```

the if block is executed, even with the same values

**Comment by thetutlage** [comment]
This line of code `if (id !== auth.user?.id) {` has nothing to do with AdonisJS. It is a Javascript expression + conditional.

Also, the root cause can be that one is a number and other is a string. So its always nice to debug the value types along with the value. Try changing it to `if (Number(id) !== auth.user?.id)`

**Reply by leandrodavimg** [reply]
It worked for me.

My question was that when typing my breakdown I have to do the parse.
